### ISSUE TITLE
Box wrap in 2D produces nan values

### ISSUE THREAD

**Issue by bdice** [glotzerlab/freud#136]
**[Original report](https://bitbucket.org/glotzer/freud/issue/136) by James Antonaglia (Bitbucket: [jamesaan](https://bitbucket.org/jamesaan), GitHub: [jamesaan](https://github.com/jamesaan)).**

----------------------------------------

The following minimal example illustrates that freud.box.Box.wrap() does not handle wrapping correctly when the box object is set to be 2D:

```
#!python

import freud
import numpy as np

r = np.array([9,11,13],dtype=np.float32)
B = freud.box.Box(Lx=10,Ly=10,is2D=True)
print('Vector r is originally',r)
B.wrap(r)
print('Vector r is wrapped to',r)

# Try specifying Lz
r = np.array([9,11,13],dtype=np.float32)
B = freud.box.Box(Lx=10,Ly=10,Lz=10,is2D=True)
print('Now vector r is originally',r)
B.wrap(r)
print('Now vector r is wrapped to',r)
print('B.Lz =',B.getLz())
```
This was tested with the latest conda-available build of freud, tagged as version 0.6.1.

**Comment by bdice** [comment]
**Original comment by Eric Harper (Bitbucket: [harperic](https://bitbucket.org/harperic), GitHub: [harperic](https://github.com/harperic)).**

----------------------------------------

Merged in FixIssue136 (pull request #91)

Fixissue136

**Comment by bdice** [comment]
**Original comment by Eric Harper (Bitbucket: [harperic](https://bitbucket.org/harperic), GitHub: [harperic](https://github.com/harperic)).**

----------------------------------------

Merged in FixIssue136 (pull request #91)

Fixissue136

**Comment by bdice** [comment]
**Original comment by Eric Harper (Bitbucket: [harperic](https://bitbucket.org/harperic), GitHub: [harperic](https://github.com/harperic)).**

----------------------------------------

Merged in FixIssue136 (pull request #91)

Fixissue136

**Comment by bdice** [comment]
**Original comment by Eric Harper (Bitbucket: [harperic](https://bitbucket.org/harperic), GitHub: [harperic](https://github.com/harperic)).**

----------------------------------------

Merged in FixIssue136 (pull request #91)

Fixissue136
